src/granite/querying.cr

```diff
@@ -115,19 +116,22 @@ module Granite::Querying
   end
 
   def exec(clause = "")
+    switch_to_writer_adapter
```

Comment:
This looks like a good placeholder to put the trigger to check if we should be switching the connection. I see it repeated a bit, and I don't think we should assume switching to the writer so easily.

Reply:
If the user calls `exec` directly, I don't think they want to perform a write operation.

Reply:
agreed